dotCMS/src/main/java/com/dotcms/filters/interceptor/saml/SamlWebInterceptor.java

```diff
@@ -0,0 +1,682 @@
+            final String log   = new Date() + ": Successfull SAML login for Site '" + host.getHostname() + "' with IdP " +
+                    "ID: " + identityProviderConfiguration.getId() + " (" + env + ") from " + request.getRemoteAddr() + " for user: " +
+                    user.getEmailAddress();
+
+            // “$TIMEDATE: SAML login success for $host (frontend|backend) from $REQUEST_ADDR for user $username”
+            SecurityLogger.logInfo(SecurityLogger.class, this.getClass() + " - " + log);
+            Logger.info(this, log);
+        } catch (Exception e) {
+
+            Logger.error(this, e.getMessage(), e);
+        }
+    }
+
+
+    protected HttpSession renewSession(final HttpServletRequest request, final HttpSession currentSession) {
+
+        String attributeName  = null;
+        Object attributeValue = null;
+        Enumeration<String> attributesNames = null;
+        HttpSession renewSession            = currentSession;
+
+        final Map<String, Object> sessionAttributes = new HashMap<>();
+
+        if (null != currentSession && !currentSession.isNew()) {
+
+            Logger.debug(this, ()-> "Renewing the HTTP session");
+
+            attributesNames = currentSession.getAttributeNames();
+
+            while (attributesNames.hasMoreElements()) {
+
+                attributeName  = attributesNames.nextElement();
+                attributeValue = currentSession.getAttribute(attributeName);
+                Logger.debug(this, "Copying attribute '" + attributeName + "' to the new session.");
+                sessionAttributes.put(attributeName, attributeValue);
+            }
+
+            Logger.debug(this, ()->"Killing the current session");
+            currentSession.invalidate(); // kill the previous session
+
+            Logger.debug(this, ()->"Creating a new one");
+            renewSession = request.getSession(true);
+
+            for (final Map.Entry<String, Object> sessionEntry : sessionAttributes.entrySet()) {
+
+                Logger.debug(this, ()->"Adding attribute '" + sessionEntry.getKey() + "' to the new session.");
+                renewSession.setAttribute(sessionEntry.getKey(), sessionEntry.getValue());
+            }
+
+        }
+
+        return renewSession;
+    }
+
+    /**
+     * Perform the logic to get or create the user from the SAML and DotCMS If
+     * the SAML_ART_PARAM_KEY, will resolve the Assertion by calling a Resolver
+     * and will create/get/update the user on the dotcms data.
+     *
+     * @param request {@link HttpServletRequest}
+     * @param identityProviderConfiguration {@link IdentityProviderConfiguration}
+     * @return User
+     * @throws IOException
+     * @throws JSONException
+     * @throws DotDataException
+     */
+    public User getUser(final HttpServletRequest request,
+                        final IdentityProviderConfiguration identityProviderConfiguration) {
+
+        User systemUser = null;
+        User user       = null;
```

Comment:
Codacy found an issue: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)